core/src/main/java/org/frankframework/jdbc/datasource/FrankResources.java

```diff
@@ -15,191 +15,26 @@
 */
 package org.frankframework.jdbc.datasource;
 
-import java.lang.reflect.Method;
-import java.sql.Driver;
 import java.util.List;
-import java.util.Map.Entry;
-import java.util.Optional;
-import java.util.Properties;
-
-import javax.sql.DataSource;
 
 import jakarta.annotation.Nonnull;
 import jakarta.annotation.Nullable;
 
-import org.apache.commons.lang3.StringUtils;
-import org.springframework.jdbc.datasource.DriverManagerDataSource;
-
-import lombok.Getter;
 import lombok.Setter;
 
-import org.frankframework.util.AppConstants;
-import org.frankframework.util.ClassUtils;
-import org.frankframework.util.CredentialFactory;
-import org.frankframework.util.StringResolver;
-import org.frankframework.util.StringUtil;
-
 public class FrankResources {
 
-	private static final AppConstants APP_CONSTANTS = AppConstants.getInstance();
-
-	private @Setter List<Resource> jdbc;
-	private @Setter List<Resource> jms;
-	private @Setter List<Resource> mqtt;
-	private @Setter List<Resource> other;
-
-	@Getter @Setter
-	public static class Resource {
-		String name;
-		String type;
-		String url;
-		String authalias;
-		String username;
-		String password;
-		Properties properties;
-
-		@Override
-		public String toString() {
-			return "Resource ["+ name+"]";
-		}
-	}
-
-	@SuppressWarnings("unchecked")
-	public @Nullable <O> O lookup(@Nonnull String name, @Nullable Properties environment, @Nonnull Class<O> lookupClass) throws ClassNotFoundException {
-		if(name.indexOf('/') == -1) {
-			throw new IllegalStateException("no resource prefix found, expected one of [jdbc/jms/mongodb/mqtt]");
-		}
+	private @Setter List<FrankResource> resources;
 
-		Resource resource = findResource(name);
-		if(resource == null) {
-			return null; // If the lookup returns null, fail-fast to allow other ResourceFactories to locate the object.
-		}
-		if(StringUtils.isEmpty(resource.getUrl())) {
-			throw new IllegalStateException("field url is required");
+	public @Nullable FrankResource findResource(@Nonnull String name) {
+		if (resources == null) {
+			return null; // No matching resources found.
 		}
 
-		Properties properties = getConnectionProperties(resource, environment);
-		String url = StringResolver.substVars(resource.getUrl(), APP_CONSTANTS);
-		String type = StringResolver.substVars(resource.getType(), APP_CONSTANTS);
-
-		Class<?> clazz = ClassUtils.loadClass(type);
-
-		if(lookupClass.isAssignableFrom(DataSource.class) && Driver.class.isAssignableFrom(clazz)) { // It's also possible to use the native drivers instead of the DataSources directly.
-			return (O) loadDataSourceThroughDriver(clazz, url, properties);
-		}
-
-		if(lookupClass.isAssignableFrom(clazz)) {
-			return (O) createInstance(clazz, url, properties);
-		}
-
-		throw new IllegalStateException("class is not of required type ["+type+"]");
+		return resources.stream()
+				.filter(e -> name.equals(e.getName()))
+				.findFirst()
+				.orElse(null);
```

Comment:
Is het niet handiger als r`esources` een map is, met de `FrankResource#name` als de key?

Dan kunnen we gewoon `resources.get(name)` doen en als de resource er niet is, is het vanzelf `null`.

Reply:
Dat ding wordt gevuld door SnakeYaml, helaas kan ik er geen map van maken...

Reply:
OK, jammer.